### PR TITLE
Require Java 11 and Jenkins 2.361.4 or newer

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Require Java 11 and Jenkins 2.361.4 or newer

Jenkins core has required Java 11 for over a year.  As part of preparing for Java 21 support, I updated this plugin to require Java 11 and to use Jenkins 2.361.4 or newer as its minimum version.

Not ready for Java 17 or Java 21 because gmaven is used to compile groovy source files and that seems to no longer be recommended practice by the gmaven plugin maintainers.

Changes in this pull request are necessary but not sufficient for modernization of this plugin.

Based on the work in the pull requests:

* #125
* #124

Other changes that were made included:

- Update project configuration
- Upgrade project to latest Jenkins version
- Remove assertions that depend on undeclared parameters
- Require Java 11 and Jenkins 2.361.4 or newer
- Use parent pom 4.74
- Use parent pom spotbugs, not outdated findbugs
- Remove dead local store
- Add more spotbugs exclusions
- Declare a string as final to satisfy checkstyle
- Use https:// instead of git:// in scm property
- Resolve checkstyle complaints in FormValidations
- Use checkstyle 3.3.0
- Exclude generated sources from checkstyle checks
- Silence checkstyle warning about space
- Add javadoc to satisfy checkstyle
- Declare some strings final to satisfy checkstyle
- Wrap the line to satisfy checkstyle
- Remove pmd - not helpful
- Do not test with Java 17

### Testing done

Confirmed that tests pass with Java 11 on Linux and that warnings are silenced from `mvn clean verify`.

Confirmed that interactive use with Jenkins 2.414.3 is no worse than it was before this pull request.  Images are now displayed correctly in the view provided by this plugin.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes

### CODE REVIEW

**Comment by yoavain-sundaysky** [comment]
This is amazing!

I took your branch and added a fix to all icons (the PR you mentioned did not fix everything).

If you want to add the icon fix to your PR, I'm attaching both a patch file and the modified files (zip)

[Fix_all_icons.patch](https://github.com/jenkinsc
[changed-files.zip](https://github.com/jenkinsci/build-pipeline-plugin/files/13164232/changed-files.zip)
i/build-pipeline-plugin/files/13164195/Fix_all_icons.patch)

**Comment by MarkEWaite** [comment]
> This is amazing!
> 
> I took your branch and added a fix to all icons (the PR you mentioned did not fix everything).
> If you want to add the icon fix to your PR, I'm attaching a zip with a patch file and the modified files
> 
> [changed-files-and-patch.zip](https://github.com/jenkinsci/build-pipeline-plugin/files/13164286/changed-files-and-patch.zip)
> 

I would love to have you submit a new pull request that includes this pull request and your additions. I'm at a conference right now and don't have enough time to pull in your change.

**Comment by MarkEWaite** [comment]
The plugin build was successful and the icons display correctly thanks to @yoavain-sundaysky 

[build-pipeline-plugin.zip](https://github.com/jenkinsci/build-pipeline-plugin/files/13193808/build-pipeline-plugin.zip)

**Comment by MarkEWaite** [comment]
@dalvizu , @recena , or @geoffbullen would you be willing to merge this?  The plugin is popular enough that I believe it should eventually be tested with Java 21.  That's not this pull request, but might arrive in a future pull request.

**Comment by dalvizu** [comment]
I'll review and merge this by Friday. I can't be an effective maintainer of this plugin going forward, is there an area to list a plugin as open for adoption?

**Comment by MarkEWaite** [comment]
> I'll review and merge this by Friday. I can't be an effective maintainer of this plugin going forward, is there an area to list a plugin as open for adoption?

Thanks for asking.  There is a place to announce that the plugin is up for adoption.  If you add the "adopt-this-plugin" topic to the GitHub repository, that will make it visible to others that it is up for adoption.  You can see an example with the [jaxb plugin](https://github.com/jenkinsci/jaxb-plugin).  As an added step, you could also announce that it is up for adoption on the [Jenkins developer mailing list](https://groups.google.com/g/jenkinsci-dev).

**Comment by dalvizu** [comment]
Released in 2.0.0

File: Jenkinsfile [hunk 1]
```diff
@@ -1,4 +1,9 @@
 #!/usr/bin/env groovy
 
 /* `buildPlugin` step provided by: https://github.com/jenkins-infra/pipeline-library */
-buildPlugin()
+buildPlugin(configurations: [
+    [ platform: 'linux', jdk: '11' ],
+    [ platform: 'windows', jdk: '11' ],
+    // Compilation fails on Java 17
+    // [ platform: 'linux', jdk: '17' ],
```
Comment (basil) [inline Jenkinsfile:8]: #133

**Pull request merged**